lib/jekyll/commands/compose.rb

```diff
@@ -0,0 +1,111 @@
+# frozen_string_literal: true
+
+module Jekyll
+  module Commands
+    class ComposeCommand < Command
+      def self.init_with_program(prog)
+        prog.command(:compose) do |c|
+          c.syntax "compose [--post|--draft|--collection COLLECTION] NAME"
+          c.description "Creates a new document with the given NAME"
+
+          options.each { |opt| c.option *opt }
+
+          c.action { |args, options| process args, options }
+        end
+      end
+
+      def self.options
+        [
+          ["extension", "-x EXTENSION", "--extension EXTENSION", "Specify the file extension"],
+          ["layout", "-l LAYOUT", "--layout LAYOUT", "Specify the document layout"],
+          ["force", "-f", "--force", "Overwrite a document if it already exists"],
+          ["date", "-d DATE", "--date DATE", "Specify the document date"],
+          ["collection", "--collection COLLECTION", "Specify the document collection"],
+          ["post", "--post", "Create a new post (default)"],
+          ["draft", "--draft", "Create a new draft"],
+          ["config", "--config CONFIG_FILE[,CONFIG_FILE2,...]", Array, "Custom configuration file"],
+        ]
+      end
+
+      def self.process(args = [], options = {})
+        config = configuration_from_options(options)
+        params = ComposeCommandArgParser.new args, options, config
+        params.validate!
+
+        document = ComposeCommandFileInfo.new params
+
+        file_creator = Compose::FileCreator.new(document, params.force?, params.source)
+        file_creator.create!
+
+        Compose::FileEditor.bootstrap(config)
+        Compose::FileEditor.open_editor(file_creator.file_path)
+      end
+
+      class ComposeCommandArgParser < Compose::ArgParser
+        def validate!
+          if [options["collection"], options["post"], options["draft"]].reject(&:nil?).length > 1
+            raise ArgumentError, "You can only specify one of --post, --draft, or --collection COLLECTION."
+          end
+
+          super
+        end
+
+        def date
+          options["date"].nil? ? Time.now : Date.parse(options["date"])
+        end
+
+        def collection
+          if !options["collection"].nil?
+            options["collection"]
+          elsif !options["post"].nil?
+            "posts"
+          elsif !options["draft"].nil?
+            "drafts"
+          else
+            "posts"
+          end
+        end
```

Comment:
Instead of *negative nil-checks*, I wonder if we could just use binary operations. For example:
```ruby
options["collection"] || options["post"] || options["draft"] || "posts"
```